iam/api/QuickStartNew/QuickStartNew.cs

```diff
@@ -0,0 +1,161 @@
+        //Role to be granted
+        var role = "roles/logging.logWriter";
+        //All permissions contained in the role to be granted
+        var rolePermissions = new List<string>() { "logging.logEntries.create" };
+        // Initialize service
+        var crmService = InitializeService();
+
+        // Grant your member the "Log writer" role for your project
+        AddBinding(crmService, projectId, member, role);
+        // Test if the member has the permissions granted by the role
+        var grantedPermissions = new List<string>(TestPermissions(crmService, projectId, member, rolePermissions));
+        // Print the permissios from the role permissions that the member actually has
+        foreach (var p in grantedPermissions)
+        {
+            Console.WriteLine(p.ToString());
+        }
+        // Remove member from the "Log writer" role
+        RemoveMember(crmService, projectId, member, role);
+    }
+
+    public static CloudResourceManagerService InitializeService()
+    {
+        // Get credentials
+        var credential = GoogleCredential.GetApplicationDefault()
+            .CreateScoped(IamService.Scope.CloudPlatform);
+
+        // Create the Cloud Resource Manager service object
+        var crmService = new CloudResourceManagerService(
+            new CloudResourceManagerService.Initializer
+            {
+                HttpClientInitializer = credential
+            });
+
+        return crmService;
+    }
+
+    public static void AddBinding(
+        CloudResourceManagerService crmService, string projectId, string member, string role)
+    {
+        // Get the project's policy by calling the
+        // Cloud Resource Manager Projects API
+        var policy = crmService.Projects.GetIamPolicy(
+            new GetIamPolicyRequest(),
+            projectId).Execute();
+
+        // Find binding in policy
+        var binding = policy.Bindings.FirstOrDefault(x => x.Role == role);
+
+        // If binding already exists, add member to binding
+        if (binding != null)
+        {
+            binding.Members.Add(member);
+
+            // If binding does not exist, add binding to policy
+        }
+        else
+        {
+            binding = new Binding
+            {
+                Role = role,
+                Members = new List<string> { member }
+            };
+            policy.Bindings.Add(binding);
+        }
+
+        // Set the project's policy by calling the
+        // Cloud Resource Manager Projects API
+        crmService.Projects.SetIamPolicy(
+           new SetIamPolicyRequest
+           {
+               Policy = policy
+           }, projectId).Execute();
+    }
+
+
+    public static IList<string> TestPermissions(
+        CloudResourceManagerService crmService, string projectId, string member, List<string> permissions)
+    {
+        TestIamPermissionsRequest requestBody = new TestIamPermissionsRequest();
+        requestBody.Permissions = new List<string>(permissions);
```

Comment:
```suggestion
        TestIamPermissionsRequest requestBody = new TestIamPermissionsRequest
        {
            Permissions = new List<string>(permissions)
        };
```
Better use object initializers.
And you don't have to create a new list, permissions is already a List<string>